token/cli/src/main.rs

```diff
@@ -5810,6 +5839,251 @@ mod tests {
+                source_keypair_file.path().to_str().unwrap(),
+                "--owner",
+                &unknown_owner.pubkey().to_string(),
+                "--fee-payer",
+                permanent_delegate_keypair_file.path().to_str().unwrap(),
+                "--program-id",
+                &config.program_id.to_string(),
+            ],
+        )
+        .await;
+
+        result.unwrap();
+
+        let ui_amount = 100.0;
+
+        mint_tokens(&config, &payer, token, ui_amount, source.pubkey()).await;
+
+        let ui_account = config
+            .rpc_client
+            .get_token_account(&source.pubkey())
+            .await
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(ui_account.token_amount.amount, "100");
+
+        let result = exec_test_cmd(
+            &config,
+            &[
+                "spl-token",
+                CommandName::Burn.into(),
+                &source.pubkey().to_string(),
+                "10",
+                "--owner",
+                permanent_delegate_keypair_file.path().to_str().unwrap(),
+                "--fee-payer",
+                permanent_delegate_keypair_file.path().to_str().unwrap(),
+                "--program-id",
+                &config.program_id.to_string(),
+            ],
+        )
+        .await;
+
+        result.unwrap();
+
+        let ui_account = config
+            .rpc_client
+            .get_token_account(&source.pubkey())
+            .await
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(ui_account.token_amount.amount, "90");
+    }
+
+    #[tokio::test]
+    #[serial]
+    async fn transfer_with_permanent_delegate() {
+        let (test_validator, payer) = new_validator_for_test().await;
+        let config =
+            test_config_with_default_signer(&test_validator, &payer, &spl_token_2022::id());
+
+        let token_keypair = Keypair::new();
+        let token = token_keypair.pubkey();
+        let bulk_signers: Vec<Arc<dyn Signer>> =
+            vec![Arc::new(clone_keypair(&payer)), Arc::new(token_keypair)];
+
+        command_create_token(
+            &config,
+            TEST_DECIMALS,
+            token,
+            payer.pubkey(),
+            false,
+            false,
+            false,
+            true,
+            None,
+            None,
+            None,
+            None,
+            bulk_signers,
+        )
+        .await
+        .unwrap();
+
+        let source = Keypair::new();
+        let destination = Keypair::new();
+
+        let permanent_delegate_keypair_file = NamedTempFile::new().unwrap();
+        write_keypair_file(&payer, &permanent_delegate_keypair_file).unwrap();
+
+        let source_keypair_file = NamedTempFile::new().unwrap();
+        write_keypair_file(&source, &source_keypair_file).unwrap();
+
+        let destination_keypair_file = NamedTempFile::new().unwrap();
+        write_keypair_file(&destination, &destination_keypair_file).unwrap();
```

Comment:
nit: Same here. Unless I'm missing something, you can probably use  the `create_associated_account` helpers rather than go through this